pkg/controller/controller_sync_task.go

```diff
@@ -156,7 +167,78 @@ func (st *syncTask) evalSpec(res *smith_v1.Resource) (*unstructured.Unstructured
 			BlockOwnerDeletion: &trueRef,
 		})
 	}
-	spec.SetOwnerReferences(refs)
+	obj.SetOwnerReferences(refs)
+
+	return obj, nil
+}
+
+// evalPluginSpec evaluates the plugin resource specification and returns the result.
+func (st *syncTask) evalPluginSpec(res *smith_v1.Resource) (*unstructured.Unstructured, error) {
+	plugin, ok := st.plugins[res.PluginName]
+	if !ok {
+		return nil, errors.Errorf("no such plugin %q", res.PluginName)
+	}
+	dependencies := make(map[smith_v1.ResourceName]smithPlugin.Dependency, len(res.DependsOn))
+	for _, name := range res.DependsOn {
+		var dependency smithPlugin.Dependency
+		unstructuredActual := st.readyResources[name]
+		gvk := unstructuredActual.GroupVersionKind()
+		actual, err := st.scheme.New(gvk)
+		if err != nil {
+			return nil, err
+		}
+		if err = unstructured_conversion.DefaultConverter.FromUnstructured(unstructuredActual.Object, actual); err != nil {
+			return nil, err
+		}
+		dependency.Actual = actual
+		switch obj := actual.(type) {
+		case *sc_v1b1.ServiceBinding:
+			secret, exists, err := st.store.Get(core_v1.SchemeGroupVersion.WithKind("Secret"), obj.Namespace, obj.Spec.SecretName)
```

Comment:
extract this block into a separate method?
also probably would be nice to have a single file with one such method per object type.

Reply:
This is a good idea

Reply:
But you didn't make this change before merging. Sigh...

Reply:
I did?

Reply:
@ash2k apologies, all good. I looked at the wrong revision I guess.